src/pages/Settings/index.tsx

```diff
@@ -0,0 +1,158 @@
+import { getGardenByGardenId } from "../../helpers/api/gardens/getGardenByGardenId";
+// import { useUserState } from "../../store/user/useUserState";
+import { useApi } from "../../utils/api/useApi";
+import { DescriptionInput } from "./components/DescriptionInput";
+import { GardenDataWrapper } from "./components/GardenDataWrapper";
+import { NameInput } from "./components/NameInput";
+import styles from "./Settings.module.css";
+
+// const useStyles = makeStyles((theme: Theme) =>
+//   createStyles({
+//     root: {
+//       maxWidth: 845,
+//     },
+//     media: {
+//       height: 140,
+//     },
+//     nameInput: {
+//       width: "100%",
+//     },
+//   })
+// );
+
+export const MyNiwaSettings = () => {
+  // const history = useHistory();
+  // const { userData } = useUserState();
+  // const classes = useStyles();
+  const { gardenId } = useParams<{ gardenId: string }>();
+
+  const [gardenDataApi, getGardenData] = useApi(getGardenByGardenId);
+
+  const garden = useMemo(() => gardenDataApi.response?.garden, [gardenDataApi]);
+  const initialGardenName = garden?.name ?? "";
+  const initialGardenDescription = garden?.description ?? "";
+
+  const rules = useMemo(
+    () => gardenDataApi.response?.rules ?? [],
+    [gardenDataApi]
+  );
+
+  // Garden Name Input
+  const [showNameInput, setShowNameInput] = useState(false);
+  const [gardenName, setGardenName] = useState<string | undefined>(undefined);
+  const currentGardenName = useMemo(() => {
+    return (gardenName !== undefined ? gardenName : garden?.name) ?? "-";
+  }, [gardenName, garden?.name]);
+
+  // Garden Description Input
+  const [showDescriptionInput, setShowDescriptionInput] = useState(false);
+  const [gardenDescription, setGardenDescription] = useState<
+    string | undefined
+  >(undefined);
+  const currentGardenDescription = useMemo(() => {
+    return (
+      (gardenDescription !== undefined
+        ? gardenDescription
+        : garden?.description) ?? "-"
+    );
+  }, [gardenDescription, garden?.description]);
+
+  const onNameInputChange = useCallback(
+    (e: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>) => {
+      const { value } = e.currentTarget;
+
+      setGardenName(value);
+    },
+    []
+  );
+
+  const onDescriptionInputChange = useCallback(
+    (e: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>) => {
+      const { value } = e.currentTarget;
+
+      setGardenDescription(value);
+    },
+    []
+  );
+
+  useEffect(() => {
+    if (gardenId) {
+      const dateISO: string = formatISO(new Date(), {
+        representation: "date",
+      });
+      getGardenData(gardenId, dateISO);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [gardenId]);
+
+  return (
+    <>
+      <Head title="Settings ⚙️" />
+      <motion.div
+        initial={{ opacity: 0 }}
+        animate={{ opacity: 1 }}
+        transition={{ duration: 0.3 }}
+        exit={{ opacity: 0 }}
+      >
+        <Section>
```

Comment:
Added name / description input, I'll keep adding for rules and category in the next PR